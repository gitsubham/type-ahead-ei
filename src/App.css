

.make-relative{
  position: relative;
}
.make-flex{
  display: flex;
}

.m100 {
  margin: 100px;
}
.options-panel {
    border: 1px solid #dedede;
    box-shadow: 0 -2px 4px 0 rgba(0, 0, 0, 0.15);
    position: absolute;
    width: 100%;
    left: 0;
    top: 100%;
    z-index: 1;
    background: #fff;
}
.options-panel-item{
  padding: 10px;
  cursor: pointer;
}
.options-panel-item:hover{
  background-color: #9ecff9;;
}
.input-control{
  display: flex;
  flex: 1;
}
.input-control input{
  border: none;
  padding: 8px;
  font-size: 14px;
  flex: 1;
}
.input-control input:focus{
  outline: none;
}

.selected-item{
  display: inline-block;
  padding: 5px;
  background: #ccc;
  border-radius: 20px;
  margin-right: 5px;
  font-size: 14px;
}
.autocomplete-wrapper{
  display: flex;
  border: 1px solid #dedede;
  padding: 5px;
  border-radius: 20px;
  align-items: center;
}
.autocomplete-wrapper:hover {
  border-color:#fb0000;
}

.autocomplete-input-section{
  position: relative;
  display: flex;
  justify-content: space-between;
  flex: 1;
  align-items: center;
  font-size:14px;
}

.f12 {
  font-size: 12px;
}

.m12 {
    margin-right: 12px;
}

.option-panel-msg {
  left: 40%;
  position: relative;
  padding: 1%;
  color: #757575;
}

.overlay {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
}


.remove-selected-option:hover {
  background-color: #FFBDAD;
  color: #DE350B;
}

.clear-all {
  text-transform: uppercase;
  color: #ea3e3e;
  font-weight: 600;
} 

