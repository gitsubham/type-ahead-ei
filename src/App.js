import React from 'react';
import { connect } from 'react-redux'; 
import { get } from 'lodash';

import './App.css';
import '../src/style.css'
import 'bootstrap/dist/css/bootstrap.css'

// import Root from './Root'
import MovieTypeAhead from './containers/MovieTypeAhead'

export default class App extends React.Component {
  render(){
    return <div className={"m100"}> <MovieTypeAhead /> </div>
  }
}
