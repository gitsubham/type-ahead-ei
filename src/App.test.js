import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import App from './App';
import Search from './components/common/Select'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

const options = [
  {
    "label": "Hong Kil-dong",
    "value": "tt0254387",
    "year": "1986"
  },
  {
    "label": "Kil",
    "value": "tt3045760",
    "year": "2013"
  },
  {
    "label": "Gyeongmajang ganeun kil",
    "value": "tt0104655",
    "year": "1991"
  },
  {
    "label": "Sampoganeun kil",
    "value": "tt0307406",
    "year": "1975"
  },
  {
    "label": "Mixed-Up Hong Kil-dong",
    "value": "tt0381723",
    "year": "1990"
  },
  {
    "label": "Hong Kil-dong",
    "value": "tt0327306",
    "year": "1967"
  }
]

describe('Search Component', () => {
  test("Search component renders with no search results", () => {
  	const wrapper = shallow(<Search options={[]} />)
  	expect(wrapper.exists()).toBe(true)
  })

  test("Search component renders with search results", () => {
  	const wrapper = shallow(<Search options={options} />)
  	expect(wrapper).toMatchSnapshot();
  })

  test("Search component search text is echoed", () => {
  	const wrapper = shallow(<Search options={[]} isSearchable={true} handleInputChange={()=>{}}/>)
  	wrapper.find("input").simulate("change", {
  		target: { value: "Fast" }
  	})
  	expect(wrapper.find("input").props().value).toEqual("Fast")
  })

  
  test("Search component search text is echoed", () => {
  	const wrapper = shallow(<Search options={[]} isSearchable={true} handleInputChange={()=>{}}/>)
  	wrapper.find("input").simulate("change", {
  		target: { value: "Fast" }
  	})
  	expect(wrapper.find("input").props().value).toEqual("Fast")
  })

});

