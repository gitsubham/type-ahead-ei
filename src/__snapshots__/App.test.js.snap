// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Search component renders with search results 1`] = `ShallowWrapper {}`;
